### PR TITLE
Add ct_mongoose_log_hook.erl to export MongooseIM into CT reports

### PR DESCRIPTION
**Pull request by arcusfelis**
Creates links to log file.
Example page is here http://arcusfelis.github.io/mongooseim-ct-reports/1208.4/ct_run.test@testing-gce-b55750a2-2995-4b0e-b62b-238580b1fcb0.2016-01-21_19.57.53/ejabberd_tests.tests.bosh_SUITE.logs/run.2016-01-21_19.58.25/bosh_suite.cant_send_invalid_rid.html

### CODE REVIEW

**Comment by lavrin** [comment]
Can you add 2 lines of description of what this hook does?

**Comment by arcusfelis** [comment]
@lavrin This PR copies ejabberd's log files into ct_report. It also creates links from test case pages to the log. So, it helps to match log events with test cases.

**Comment by michalwski** [comment]
This looks promising. @arcusfelis will you have some time to rebase?

**Comment by arcusfelis** [comment]
Revived from the dead PR.
Overall simplification.
Adding custom HTML code into CT reports is getting a bit too hard.

**Comment by arcusfelis** [comment]
Updated test result example http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/645/3001/internal_mnesia.18.3/big/ct_run.test@testing-gce-590f0ff5-8bb7-46d8-bbf9-162b227081b3.2017-03-31_12.24.14/ejabberd_tests.tests.bosh_SUITE.logs/run.2017-03-31_12.25.22/bosh_suite.interleave_requests_statem.586.html

File: test.disabled/ejabberd_tests/tests/adhoc_SUITE.erl [hunk 1]
```diff
@@ -29,7 +29,7 @@ groups() ->
     [{adhoc, [sequence], [ping]}].
 
 suite() ->
-    escalus:suite().
+    [{timetrap,{seconds,30}}|escalus:suite()].
```
Comment (michalwski) [inline test.disabled/ejabberd_tests/tests/adhoc_SUITE.erl]: Why exactly do we need these `timetrap`s in all suites? Could it be configured globally from the `default.spec` file?

Comment (fenek) [inline test.disabled/ejabberd_tests/tests/adhoc_SUITE.erl]: +1
What is the actual benefit of adding these timetraps at all?

File: test.disabled/ejabberd_tests/src/ct_mongoose_log_hook.erl [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+
+%% Does not always work. One host can be localhost, another one myhost.
+compare_host_names(Node1, Node2) ->
+    Host1 = node_to_host(Node1),
+    Host2 = node_to_host(Node2),
+    Host1 =:= Host2.
+
+node_to_host(Node) when is_atom(Node) ->
+    [$@|Host] = lists:dropwhile(fun(X) -> X =/= $@ end, atom_to_list(Node)),
```
Comment (fenek) [inline test.disabled/ejabberd_tests/src/ct_mongoose_log_hook.erl]: Why not just `string:tokens/2`?

File: test.disabled/ejabberd_tests/tests/adhoc_SUITE.erl (hunk 1 above)
Comment (arcusfelis) [inline test.disabled/ejabberd_tests/tests/adhoc_SUITE.erl]: Timetraps are used inside the hook module.
If we stuck inside a hook module, than we would have **very** strange behaviour (most likely the travis task would be timeout).
We can use a default timetrap, but than we still need to change the default trap for bosh and mam modules.

**Comment by arcusfelis** [comment]
This PR build **have** just failed because of some bug in SM, so you can see this feature in action:

http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/645/3197/internal_mnesia.18.3/big/ct_run.test@testing-gce-3a45af8d-a53d-4128-8aa3-ecc9c6a0935d.2017-05-26_11.27.30/ejabberd_tests.tests.sm_SUITE.logs/run.2017-05-26_11.43.46/sm_suite.preserve_order.2138.html

It created a link to http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/645/3197/internal_mnesia.18.3/big/ct_run.test@testing-gce-3a45af8d-a53d-4128-8aa3-ecc9c6a0935d.2017-05-26_11.27.30/mongooseim@localhost.log.html#L282

Not sure how relevant it is. I suggest to extent "resumption error" error message with more information.

**Comment by arcusfelis** [comment]
And another http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/645/3197/cassandra_mnesia.18.3/big/ct_run.test@testing-gce-13dbbb97-9ba6-4085-8ddc-c87eea1b6d98.2017-05-26_12.02.40/ejabberd_tests.tests.pep_SUITE.logs/run.2017-05-26_12.17.02/pep_suite.send_caps_after_login_test.1627.html

**Pull request merged**